### PR TITLE
add more coverage to sd_config

### PR DESCRIPTION
**Pull request by meilu-git**

### CODE REVIEW

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs [hunk 1]
```diff
@@ -23,6 +23,20 @@ use super::sata_thumbprint;
...
+        assert!(
+            !$resp.pta_csr.is_empty(),
+            "PTACSR must be present after successful PartInit"
+        );
+        assert!(
+            !$resp.pta_report.is_empty(),
+            "PTAReport must be present after successful PartInit"
+        );
+    };
+}
```
[the remark this answers is not in the document]
Comment (meilu-git) [reply ddi/tbor/types/tests/commands/part_init/sd_config.rs:40]: resolved per suggestion

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs [hunk 2]
```diff
@@ -50,17 +63,1078 @@ fn part_init_with_sapota_emu() {
...
+            &known_good_part_policy(),
+            &pota_thumbprint(),
+            &shared_thumbprint,
+            Some(&shared_thumbprint),
+        )
+        .expect("PartInit with matching SATA and SAPOTA values must succeed");
+
+    assert!(!resp.pta_csr.is_empty(), "PTACSR must be non-empty");
+    assert!(!resp.pta_report.is_empty(), "PTAReport must be non-empty");
+}
```
[the remark this answers is not in the document]
Comment (meilu-git) [reply ddi/tbor/types/tests/commands/part_init/sd_config.rs]: resolved per suggestion

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs [hunk 3]
```diff
@@ -50,17 +63,1078 @@ fn part_init_with_sapota_emu() {
...
+            &mach_seed(),
+            &known_good_part_policy(),
+            &pota_thumbprint(),
+            &sata_thumbprint(),
+            None,
+        )
+        .expect_err("second PartInit without erase must be rejected");
+
+    eprintln!("second PartInit rejection: {err:?}");
+}
```
[the remark this answers is not in the document]
Comment (meilu-git) [reply ddi/tbor/types/tests/commands/part_init/sd_config.rs:1063]: resolved per suggestion

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs [hunk 4]
```diff
@@ -50,17 +63,1078 @@ fn part_init_with_sapota_emu() {
...
+            &session,
+            &mach_seed(),
+            &known_good_part_policy(),
+            &pota_thumbprint(),
+            &sata_thumbprint(),
+            None,
+        )
+        .expect_err("PartInit with a session created before erase must be rejected");
+
+    eprintln!("stale-session PartInit rejection: {err:?}");
```
[the remark this answers is not in the document]
Comment (meilu-git) [reply ddi/tbor/types/tests/commands/part_init/sd_config.rs]: resolved per suggestion

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs [hunk 5]
```diff
@@ -50,17 +65,1019 @@ fn part_init_with_sapota() {
...
+            &shared_thumbprint,
+            Some(&shared_thumbprint),
+        )
+        .expect("PartInit must accept matching SATA and SAPOTA byte values");
+
+    assert_part_init_artifacts_present!(resp);
+}
+
+#[test]
+fn part_init_with_boundary_pattern_sd_thumbprints_emu() {
```
Comment (rajesh-gali) [inline ddi/tbor/types/tests/commands/part_init/sd_config.rs]: please remove _emu from all test names

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs [hunk 6]
```diff
@@ -50,17 +65,1019 @@ fn part_init_with_sapota() {
...
+
+    let resp = ctx
+        .part_init_sd(&session, &seed, &policy, &pota, &sata, Some(&sapota))
+        .expect("PartInit with independently modified merged SD inputs must succeed");
+
+    assert_part_init_artifacts_present!(resp);
+}
+
+#[test]
+fn part_init_sd_without_sapota_still_returns_both_artifacts_emu() {
```
Comment (rajesh-gali) [inline ddi/tbor/types/tests/commands/part_init/sd_config.rs]: Add a negative test for non-empty SAPOTA with length ≠ 48

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs [hunk 7]
```diff
@@ -50,17 +65,1019 @@ fn part_init_with_sapota() {
...
+    );
+    assert!(
+        resp.pta_report.len() > 1,
+        "PTAReport must contain encoded data, not only an empty placeholder"
+    );
+}
+
+/// Verifies that PartInit accepts different valid SATA thumbprints across clean device states.
+#[test]
+fn part_init_with_different_sata_thumbprints_emu() {
```
Comment (rajesh-gali) [inline ddi/tbor/types/tests/commands/part_init/sd_config.rs]: Add a test asserting different thumbprints produce different report_data

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs [hunk 8]
```diff
@@ -50,17 +65,1019 @@ fn part_init_with_sapota() {
...
+            &sata_thumbprint(),
+            None,
+        )
+        .expect_err("PartInit with a session created before erase must be rejected");
+
+    assert!(
+        matches!(
+            err,
+            azihsm_ddi_interface::DdiError::TborStatus(_)
+                | azihsm_ddi_interface::DdiError::DdiError(_)
```
Comment (rajesh-gali) [inline ddi/tbor/types/tests/commands/part_init/sd_config.rs]: can we make this to assert on specific error?

Comment (meilu-git) [reply ddi/tbor/types/tests/commands/part_init/sd_config.rs]: resolved per suggestions

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs (hunk 7 above)
Comment (meilu-git) [reply ddi/tbor/types/tests/commands/part_init/sd_config.rs]: added per suggestion

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs (hunk 6 above)
Comment (meilu-git) [reply ddi/tbor/types/tests/commands/part_init/sd_config.rs]: added per suggestion

File: ddi/tbor/types/tests/commands/part_init/sd_config.rs (hunk 5 above)
Comment (meilu-git) [reply ddi/tbor/types/tests/commands/part_init/sd_config.rs]: removed per suggestion

**Review by rajesh-gali — APPROVED** [review]
